+sw_tests/+unit_tests/unittest_spinw_field.m

```diff
@@ -0,0 +1,59 @@
+classdef unittest_spinw_field < sw_tests.unit_tests.unittest_super
+    methods (Test)
+        function test_incorrect_shape_field_raises_error(testCase)
+            swobj = spinw();
+            testCase.verifyError(...
+                @() field(swobj, [0; 0]), ...
+                'spinw:magfield:ArraySize');
+        end
+        function test_default_field(testCase)
+            swobj = spinw();
+            testCase.assertEqual(field(swobj), [0 0 0]);
+        end
+        function test_set_field(testCase)
+            swobj = spinw();
+            field_val = [1 2 3];
+            field(swobj, field_val);
+            testCase.assertEqual(field(swobj), field_val);
+        end
+        function test_set_field_column_vector(testCase)
+            swobj = spinw();
+            field_val = [1; 2; 3];
+            field(swobj, field_val);
+            testCase.assertEqual(field(swobj), field_val');
+        end
+        function test_set_field_multiple_times(testCase)
+            swobj = spinw();
+            field(swobj, [1 2 3]);
+            new_field_val = [1.5 2.5 -3.5];
+            field(swobj, new_field_val);
+            testCase.assertEqual(field(swobj), new_field_val);
+        end
+        function test_returns_spinw_obj(testCase)
```

Comment:
It seems odd that a method of `spinw` would return a `spinw` object (when a new field is set), are there other methods that do this? Even more confusing 9to me at least) that it then acts like a getter if no field is set...

Reply:
Yeah, it is a bit odd, I don't know of other functions that do this (yet). I've added a note of this to the 'More consistent syntax/naming/shape/typing' draft issue in the project board to remind us to think about this...

Actually it returns a handle to the original spinw object so I'll change this check to check that it is a handle to the same object.